Expensify/App
# Network: Persisted Requests Queue - clear persisted requests after the response

### Details

Additional functionality was added to `Network.js` and `NetworkRequestQueue.js` to better handle persisted requests
- clear requests from persistent storage only after they are completed
- retry persisted request

Added unit tests covering persisted request handling

### Fixed Issues

$ https://github.com/Expensify/App/issues/5987

### Tests


1. Launch the app
2. Go offline
3. Write a few messages
4. Close the app
5. Be online
6. Open the app
7. Observe the messages sent while being offline are correctly sent to the backend

### QA Steps


Follow the steps from the bug report

1. Make sure you are offline.
2. Send some messages to any user.
3. Refresh the page on the web. Close & open the app on mobile.
4. Now you can still see the messages.
5. Now enable internet
6. Shortly (a second after resuming the connection) after refresh or close the app
7. The messages should be posted after the refresh or after you open the app again
  - note: the order of the messages might be incorrect, this PR is not addressing the order

This comment has additional details on steps and timing: https://github.com/Expensify/App/issues/5987#issuecomment-962959408

### Tested On


### Screenshots


#### Web

https://user-images.githubusercontent.com/12156624/144221544-5308bdb4-902b-4f5f-acb1-9564ac637777.mp4

#### Mobile Web

https://user-images.githubusercontent.com/12156624/144224565-d756ee4b-d2d4-4c50-b35e-fb5b1ebce8c9.mp4

#### Desktop

https://user-images.githubusercontent.com/12156624/144230141-1c63ea16-429f-4544-b949-4cbd1a7c968f.mov

#### iOS

https://user-images.githubusercontent.com/12156624/144236936-5b52e27b-cb66-4cde-8272-17073c9fa35d.mov

#### Android

https://user-images.githubusercontent.com/12156624/144220867-0adf287f-c63d-4220-8a94-0a2fa5ea89c5.mp4
